.pre-commit-hooks.yaml

```diff
@@ -0,0 +1,11 @@
+- id: pip-compile
+  name: pip-compile
+  description: "Automatically run 'uv pip compile' on your requirements"
+  entry: uv pip compile
+  language: python
+  files: ^requirements\.(in|txt)$
```

Comment:
Does this preclude use of files like `dev-requirements.txt` and `requirements-dev.txt`?

Reply:
Does this also exclude use of `pyproject.toml` as a source?

Reply:
(We should probably add an example for these to the README)

Reply:
I can make it handle more cases. It just depends on what is considered "too" much. Because typically, people will have different files they want the hook to watch. For example `pyproject.toml`. 
Just advise

Reply:
Yeah I'm not sure. We can probably tweak this as we learn more about use-cases? I'm okay with keeping it simple to start.